### ISSUE TITLE
v2.1.19 - Can't login on iOS, Register menu also does not work

### ISSUE THREAD

**Issue by AdrienM76** [geoffhumphrey/brewcompetitiononlineentry#1184]
Version:2.1.19

Installation URL:bataille.deadyeast.org

Is your installation hosted on brewcompetition.com or brewcomp.com?no

Description of Issue:
Trying with both Safari and Firefox on iOS on an iPad Pro, I can not log in. The 'Log In' link in the upper right corner does nothing. It appears this is some sort of overlay or modal dialog that only appears to work on desktop. Is there no direct link to a login page?

The 'Register' menu item which has a drop down also does nothing on iOS.

Perhaps these are using JS? For a simple menu or a login?

**Comment by AdrienM76** [comment]
Apologies. I should have investigated more. It appears Firefox Focus was enabled as a Content Blocker for Safari on iOS, which was preventing any drop downs, overlays, modals, buttons, etc. from working. But still, most of that functionality should not so susceptible to being blocked. Participants and Admins shouldn't have to turn off such preferences for basic functionality.

**Comment by AdrienM76** [comment]
Understood. Certainly, the warning will help others who stumble on such issues if they block JS.

Thanks for the non-modal log-in link. (and the additional links for 'My Account' and 'Log Out' on the way) I was sure it existed but wasn't able to find it at the time. (I was in a hurry to pick up beer for our comp and didn't get to investigate the files on the server.)

Since I found all sorts of things don't work with that blocker enabled, I'll just leave it off and access my comp instance from Safari.

I appreciate the quick response and adjustments made for the next version!

**Comment by AdrienM76** [comment]
After more poking around, I discovered some more 'workaround' or 'solution.' It seems Safari on iOS has two settings for Content Blockers, one near the top of the setting list where you turn individual blockers on or off, then another near the bottom where you can set them to be active on every site. (or allegedly only certain sites, like a blacklist, but that can't appear to be edited and it doesn't populate on its own oddly)

If I edit the Firefox Focus settings (from within the app, not from within iOS settings) there are options to block web fonts, that needs to be turned off to see things like printer icon and down arrows among probably other tid bits. Then there is an option for 'Other Content Blocking' and that also needs to be turned off. Then BrewComp fully works in Firefox Focus. (but still not Safari unless you turn off the setting for 'all sites' as mentioned above)

Bottom line, BrewComp works with Firefox Focus, but you have to back off some of the more aggressive blocking settings. (also sadly, not configurable per site, but I can live with it as is, this isn't BrewComp's fault, it is Mozilla's)
